src/generic_core/uproxy_core.ts

```diff
@@ -689,109 +642,53 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
-      case uproxy_core_api.CloudOperationType.CLOUD_DESTROY:
-        return this.destroyCloudServer_().then(newCloudOpResult);
-      case uproxy_core_api.CloudOperationType.CLOUD_REBOOT:
-        return this.rebootCloudServer_().then(newCloudOpResult);
-      case uproxy_core_api.CloudOperationType.CLOUD_HAS_OAUTH:
-        return this.cloudHasOAuth().then(
-          (hasOAuth :boolean) :uproxy_core_api.CloudOperationResult => {
-            return {hasOAuth: hasOAuth};
-          }
-        );
+        return this.createCloudServer_(args.region);
       default:
         return Promise.reject(new Error('cloud operation not supported'));
     }
   }
 
-  public cloudHasOAuth = () :Promise<boolean> => {
-    return oneShotModule_(CLOUD_PROVIDER_MODULE_NAME,
-                          (provider :any) => provider.hasOAuth());
-  }
-
   private createCloudServer_ = (region: string) => {
     log.debug('creating cloud server in %1', region);
 
-    // Since this step can take a while, start >0 so there's less confusion
-    // that this is a progress bar.
-    ui.update(uproxy_core_api.Update.CLOUD_INSTALL_STATUS, 'CLOUD_INSTALL_STATUS_CREATING_SERVER');
-    ui.update(uproxy_core_api.Update.CLOUD_INSTALL_PROGRESS, CLOUD_DEPLOY_PROGRESS / 2);
-
-    return this.loginIfNeeded_('Cloud').then((cloudNetwork) => {
-      return oneShotModule_(CLOUD_PROVIDER_MODULE_NAME, (provider: any) => {
-        return provider.start(CLOUD_DROPLET_NAME, region);
-      }).catch((e: any) => {
-        if (e.errcode === 'VM_AE') {
-          // This string has special meaning for the polymer template.
-          return Promise.reject(new Error('server already exists'));
-        } else {
-          return Promise.reject(e);
-        }
-      }).then((serverInfo: any) => {
-        const host = serverInfo.network.ipv4;
-        const port = serverInfo.network.ssh_port;
-
-        log.debug('installing cloud on new droplet at %1:%2 (server details: %3)', host, port, serverInfo);
-
-        ui.update(uproxy_core_api.Update.CLOUD_INSTALL_PROGRESS, CLOUD_DEPLOY_PROGRESS);
-        ui.update(uproxy_core_api.Update.CLOUD_INSTALL_STATUS, 'CLOUD_INSTALL_STATUS_LOGGING_IN');
-
-        return oneShotModule_(CLOUD_INSTALLER_MODULE_NAME, (installer: any) => {
-          installer.on('status', (status: number) => {
-            ui.update(uproxy_core_api.Update.CLOUD_INSTALL_STATUS, status);
-          });
-
-          installer.on('progress', (progress: number) => {
-            ui.update(uproxy_core_api.Update.CLOUD_INSTALL_PROGRESS,
-              CLOUD_DEPLOY_PROGRESS + (progress * ((100 - CLOUD_DEPLOY_PROGRESS) / 100)));
-          });
-
-          return installer.install(host, port, 'root', serverInfo.ssh.private);
-        }).then((cloudNetworkData: any) => {
+    const REDIRECT_URL = 'https://fmdppkkepalnkeommjadgbhiohihdhii.chromiumapp.org';
+    let oauthUrl = uproxy_cloud_install.getDigitalOceanOAuthURL(
```

Comment:
Most of these variables can be `const`, too - I find it makes things a little easier to grok when "variable variables" are explicitly called out.